app/controllers/peer_reviews_controller.rb

```diff
@@ -1,67 +1,147 @@
+    selected_reviewer_group_ids = params[:selectedReviewerGroupIds]
+    selected_reviewee_group_ids = params[:selectedRevieweeGroupIds]
+    reviewers_to_remove_from_reviewees_map = params[:selectedReviewerInRevieweeGroups]
+    action_string = params[:actionString]
+
+    if action_string == 'random_assign' or action_string == 'assign'
+      if selected_reviewer_group_ids.empty?
+        render text: t('peer_review.empty_list_reviewers'), status: 400
+        return
+      elsif selected_reviewee_group_ids.empty?
+        render text: t('peer_review.empty_list_reviewees'), status: 400
+        return
       end
     end
-  end
 
-  # PATCH/PUT /peer_reviews/1
-  # PATCH/PUT /peer_reviews/1.json
-  def update
-    respond_to do |format|
-      if @peer_review.update(peer_review_params)
-        format.html { redirect_to @peer_review, notice: 'Peer review was successfully updated.' }
-        format.json { render :show, status: :ok, location: @peer_review }
+    case action_string
+      when 'random_assign'
+        randomly_assign
+      when 'assign'
+        reviewer_groups = Grouping.where(id: selected_reviewer_group_ids)
+        reviewee_groups = Grouping.where(id: selected_reviewee_group_ids)
+        assign(reviewer_groups, reviewee_groups)
+      when 'unassign'
+        unassign(reviewers_to_remove_from_reviewees_map)
       else
-        format.html { render :edit }
-        format.json { render json: @peer_review.errors, status: :unprocessable_entity }
+        render text: t('peer_review.problem'), status: 400
+        return
+    end
+
+    head :ok
+  end
+
+  def randomly_assign
+    # TODO
+  end
+
+  def assign(reviewer_groups, reviewee_groups)
+    # TODO - Do not allow assigning if the user is already assigned (maybe put unique constraints on the table instead?)
+    reviewer_groups.each do |reviewer_group|
+      reviewee_groups.each do |reviewee_group|
+        result = reviewee_group.current_submission_used.get_latest_result
+        unless PeerReview.exists?(reviewer: reviewer_group, result: result)
+          PeerReview.create(reviewer: reviewer_group, result: result)
+        end
       end
     end
   end
 
-  # DELETE /peer_reviews/1
-  # DELETE /peer_reviews/1.json
-  def destroy
-    @peer_review.destroy
-    respond_to do |format|
-      format.html { redirect_to peer_reviews_url, notice: 'Peer review was successfully destroyed.' }
-      format.json { head :no_content }
+  def unassign(reviewers_to_remove_from_reviewees_map)
+    reviewers_to_remove_from_reviewees_map.each do |reviewee_id, reviewer_id_to_bool|
+      reviewer_id_to_bool.each do |reviewer_id, dummy_value|
+        reviewee_group = Grouping.find_by_id(reviewee_id)
+        result_id = reviewee_group.current_submission_used.get_latest_result.id
+        pr = PeerReview.find_by(result_id: result_id, reviewer_id: reviewer_id)
+        pr.destroy
+      end
     end
   end
 
+  # TODO - are these needed?
```

Comment:
Do I need to delete the following below? They may have been left over from another commit that was just a placeholder.